### PR TITLE
Get top tracks method and Factory classes

### PR DESCRIPTION
**Pull request by HussainE4**
## Motivation and Context

This PR adds the get top tracks method to act as an interface for the Spotify api call implemented by Brian. This is required for the follow artist use case. Also, adds Factory classes for entities as we are required to use 2 Design Patterns

## Your Changes

**Description**:
Added the get top tracks method in Spotify dao which takes in an Artist ID and returns an ArrayList of their Top Tracks. Added Factory classes for all the entities and updated Dao's to use them
**Type of change** (select all that apply):

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (internal change to codebase, without changing functionality)

## Checklist

- [x] I have performed a self-review of my own code.
- [x] Coding style (conforms to Google Java Style Guide)
- [x] Especially follows section 7 (Javadoc is present for every public class, public or protected member)
- [x] Especially follows section 4.6 (Multiple consecutive blank lines are not encouraged)
- [x] Correctness (follows CA and SOLID, is a reasonable and not overly complicated solution, etc.)
- [x] Description of pull request is clear, consise, and reasonably detailed

### CODE REVIEW

File: src/data_access/SpotifyDataAccessObject.java [hunk 1]
```diff
@@ -18,6 +16,7 @@ public class SpotifyDataAccessObject implements SearchUserDataAccessInterface {
 
   public SpotifyDataAccessObject(TrackDataAccessObject trackDataAccessObject,
       ArtistDataAccessObject artistDataAccessObject) {
+
```
Comment (umututku03) [inline src/data_access/SpotifyDataAccessObject.java:19]: Do not add random blank lines, it's bad practice (for possible merge conflicts).

**Review by umututku03 — CHANGES_REQUESTED** [review]
Overall perfect, just get rid of the random insertion of the blank line.

**Review by umututku03 — APPROVED** [review]
Good job!

**Review by bryjjch — APPROVED** [review]
Good adherence to the factory design patterns, and good implementation of the DAOs. Good structure and quality of code.

**Pull request merged**